apps/src/music/views/MusicView.jsx

```diff
@@ -181,6 +199,30 @@ class UnconnectedMusicView extends React.Component {
         this.player,
         this.progressManager?.getCurrentStepDetails().toolbox
       );
+      this.musicBlocklyWorkspace.addSaveEventListener(
+        ProjectManagerEvent.SaveStart,
+        () => {
+          this.props.setProjectUpdatedSaving();
+        }
+      );
+      this.musicBlocklyWorkspace.addSaveEventListener(
+        ProjectManagerEvent.SaveSuccess,
+        status => {
+          this.props.setProjectUpdatedAt(status.updatedAt);
+        }
+      );
+      this.musicBlocklyWorkspace.addSaveEventListener(
+        ProjectManagerEvent.SaveNoop,
+        status => {
+          this.props.setProjectUpdatedAt(status.updatedAt);
+        }
+      );
+      this.musicBlocklyWorkspace.addSaveEventListener(
+        ProjectManagerEvent.SaveFail,
+        () => {
+          this.props.setProjectUpdatedError();
+        }
+      );
```

Comment:
It seems this model is basically using events to try and duplicate some state that already exists elsewhere, but seems to lose the advantage of a reactive model in which a single source of state can be used by multiple components across the system.  Is there some project-management state that we can simply access in `MusicView`?

Reply:
Thanks for the offline discussion that reminded me that `ProjectManager` in its current form doesn't use Redux (by design) and so this is indeed the way that we can get its state changes into `projectRedux` where they drive the header's UI.